src/oas2/transformers/responses.ts

```diff
@@ -1,51 +1,72 @@
-import { IHttpOperationResponse } from '@stoplight/types';
+import { DeepPartial, Dictionary, IHttpOperationResponse, Optional } from '@stoplight/types';
 import { JSONSchema4 } from 'json-schema';
-import { chain, map, partial } from 'lodash';
-import { Response } from 'swagger-schema-official';
+import { chain, compact, map, partial } from 'lodash';
+import type { Response, Spec } from 'swagger-schema-official';
 
+import { isDictionary, maybeResolveLocalRef } from '../../utils';
+import { isResponseObject } from '../guards';
 import { getExamplesFromSchema } from './getExamplesFromSchema';
 import { translateToHeaderParams } from './params';
 
-function translateToResponse(produces: string[], response: Response, statusCode: string): IHttpOperationResponse {
-  const headers = translateToHeaderParams(response.headers || {});
-  const objectifiedExamples = chain(
-    response.examples || (response.schema ? getExamplesFromSchema(response.schema) : void 0),
-  )
-    .mapValues((value, key) => ({ key, value }))
-    .values()
-    .value();
-
-  const contents = produces.map(produceElement => ({
-    mediaType: produceElement,
-    schema: response.schema as JSONSchema4,
-    examples: objectifiedExamples.filter(example => example.key === produceElement),
-  }));
-
-  const translatedResponses = {
-    code: statusCode,
-    description: response.description,
-    headers,
-    contents,
-  };
+function responseTransformer(document: DeepPartial<Spec>) {
+  return function translateToResponse(
```

Comment:
Ok two observations

1. I'll keep saying that the dependencies go at the end. I still propose to reverse the order.
2. In my opinion it does not really make that much sense to just apply this pattern in a single place. Either we do (1) and we do it everywhere — or we just don't do it at all and put the `document` as a parameter.

Reply:
This allowed for nice composition with the `map` where it's used - undoing this will make the map more cumbersome - maybe it's worth it just for that?

Reply:
If I've localised the line you're referring to correctly, I do not see any improvement in the composition; I rather see a partial application which, in case we want to pursue that approach, I'd extend to the maximum extent:

1. All arguments in one line:

```ts
function responseTransformer(document: DeepPartial<Spec>, produces: string[], response: unknown, statusCode: string): Optional<IHttpOperationResponse>
```

2. Partial application for the first two arguments:

```ts
  return compact<IHttpOperationResponse>(
    map<Dictionary<unknown>, Optional<IHttpOperationResponse>>(
      responses,
      partial(responseTransformer, document, produces),
    ),
  );
```

Reply:
Oh right the oas2 one uses partial - I was thinking of the oas3 version - https://github.com/stoplightio/http-spec/blob/fix/local-references/src/oas3/transformers/responses.ts#L43.

I don't really care that much, happy to change. Given oas3 version, do you still think we should remove this pattern from here (and the oas3 spot)? Again, I don't mind either way.

Reply:
I'd use partial everywhere — it's more appropriate. We can evaluate dependency injection at a separate stage, but that (as you noticed) will require a through discussion.